### PR TITLE
[#1809] Rewrote trackevent editor to support advanced groups of manifest options

### PR DESCRIPTION
**Pull request by mjschranz**
https://webmademovies.lighthouseapp.com/projects/65733/tickets/1809-add-support-for-fontscolourssizes-to-plugins-event-editor

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @secretrobotron received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/43755ab806c1010/output.txt

File: src/editor/trackevent-editor.js [hunk 1]
```diff
@@ -256,6 +312,28 @@ define( [ "util/lang", "util/keys", "./base-editor",
...
+      }
+      else if ( manifestEntry.elem === "anchor" ) {
+        editorElement = propertyArchetype.querySelector( "a" );
+
+        editorElement.innerHTML = manifestEntry.text;
+        editorElement.title = manifestEntry.title;
+        editorElement.setAttribute( "selected", false );
+      }
+      else if ( manifestEntry.elem === "color-wheel" ) {
+        // TODO: Figure out exactly what color-wheel thingy we want to use
```
Comment (secretrobotron) [inline src/editor/trackevent-editor.js]: This makes lint angry. Can you just leave it out for now so we can merge this onto master without hassle?

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/43755ab806c1010/output.txt

Total script time: 0.75 mins

Server is running on http://pr757.butyr.org/templates/basic/

File: src/editor/trackevent-editor.js [hunk 2]
```diff
@@ -18,7 +18,23 @@ define( [ "util/lang", "util/keys", "./base-editor",
                           KeysUtils.DELETE,
                           KeysUtils.TAB,
                           KeysUtils.ESCAPE
-                        ];
+                        ],
+      __googleFonts = [];
+
+  // Load Google Fonts right off the bat. Avoid being loaded multiple times.
+  if ( __googleFonts.length === 0 ) {
```
Comment (secretrobotron) [inline src/editor/trackevent-editor.js]: If you're not pushing/popping/referencing `__googleFonts`, there's no point in initializing as an array first, right? How about just declaring `__googleFonts` and then performing simple checks for it when it's used (here, and https://github.com/mozilla/butter/pull/757/files#L1R320)?

File: src/editor/trackevent-editor.js [hunk 3]
```diff
@@ -167,6 +190,31 @@ define( [ "util/lang", "util/keys", "./base-editor",
...
     /**
+     * Member: attachAnchorChangeHandler
+     *
+     * Attaches handlers to a anchor element and updates the TrackEvent corresponding to the given property name
+     *
+     * @param {DOMElement} element: Element to which handler is attached
+     * @param {TrackEvent} trackEvent: TrackEvent to update
+     * @param {String} propertyName: Name of property to update when change is detected
+     */
+    extendObject.attachAnchorChangeHandler = function( element, trackEvent, propertyName ) {
```
Comment (secretrobotron) [inline src/editor/trackevent-editor.js]: Where does this get used? I'm assuming something in the photo/zoink editor will eventually use it, right?

File: src/editor/default.js [hunk 4]
```diff
@@ -80,7 +80,7 @@ define( [ "text!./default.html", "editor/editor" ],
                 
               }
             }
-          }, null, optionsContainer, [ 'target' ] );
+          }, null, optionsContainer, null, [ 'target' ] );
```
Comment (k88hudson) [inline src/editor/default.js]: Double quotes

File: src/layouts/trackevent-editor-defaults.html [hunk 5]
```diff
@@ -15,6 +15,14 @@
   </div>
 </div>
 
+<div class="trackevent-property anchor">
```
Comment (k88hudson) [inline src/layouts/trackevent-editor-defaults.html]: remove anchor class

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/bb31a6f90d9f650/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/bb31a6f90d9f650/output.txt

Total script time: 0.77 mins

Server is running on http://pr757.butyr.org/templates/basic/
Run the Test Suite: http://pr757.butyr.org/test/

**Pull request merged**